common/app/model/facia/PressedCollection.scala

```diff
@@ -31,6 +31,28 @@ case class PressedCollection(
   hasMore: Boolean
 ) {
 
+  lazy val isEmpty: Boolean = curated.isEmpty && backfill.isEmpty && treats.isEmpty
+
+  lazy val adFree = {
```

Comment:
Rather than a lazy val here, it might be worth just creating a new PressedCollection with a function (okay a method in a companion object because Scala). E.g.:

    def adFree(collection: PressedCollection): PressedCollection

? This PR is already done, but worth considering this kind of approach in the future. It relies on fewer language features and replaces a method that relies on 'this' with a pure function. Lazy vals are another language feature I'm not really keen on - here we are using them to avoid premature work, but a def is simpler.